src/ledger/test/LedgerTxnTests.cpp

```diff
@@ -2804,6 +2819,308 @@ TEST_CASE("Erase performance benchmark", "[!hide][erasebench]")
+
+            for (auto& e : entries)
+            {
+                resources.readBytes += xdr::xdr_size(e);
+                auto k = LedgerEntryKey(e);
+                keysToPrefetch.emplace(k);
+                // Randomly add the key to either the read or write set.
+                if (stellar::rand_flip())
+                {
+                    resources.footprint.readOnly.emplace_back(k);
+                }
+                else
+                {
+                    resources.footprint.readWrite.emplace_back(k);
+                }
+                // If the key is not a TTL and its offset is less than 0 and
+                // it has not been marked as expected to succeed in another
+                // transaction, add it to the expected failed keys set.
+                if (k.type() != TTL && offset < 0 &&
+                    expectedSuccessKeys.find(k) == expectedSuccessKeys.end())
+                {
+                    expectedFailedKeys.emplace(k);
+                }
+                else
+                {
+                    expectedSuccessKeys.emplace(k);
+                    // If a key has sufficient in one transaction, remove it
+                    // from the expected failed keys set.
+                    expectedFailedKeys.erase(k);
+                }
+            }
+            if (offset + resources.readBytes < 0)
+            {
+                resources.readBytes = 0;
+            }
+            else
+            {
+                resources.readBytes += offset;
+            }
+
+            for (auto& k : deadKeys)
+            {
+                resources.footprint.readOnly.emplace_back(k);
+            }
+
+            REQUIRE(resources.footprint.readOnly.size() +
+                        resources.footprint.readWrite.size() ==
+                    entries.size() + deadKeys.size());
+            lkMeterHot->addTxn(resources);
+            lkMeterCold->addTxn(resources);
+            ensureLedgerKeyMeterCanLoad(
+                lkMeterHot, entries, expectedSuccessKeys, expectedFailedKeys);
+            ensureLedgerKeyMeterCanLoad(
+                lkMeterCold, entries, expectedSuccessKeys, expectedFailedKeys);
+        };
+        auto checkPrefetch = [&](std::vector<LedgerEntry>& allEntries) {
+            LedgerTxn ltx2(root);
+            auto numLoadedCold =
+                root.prefetchSoroban(keysToPrefetch, lkMeterCold);
+            REQUIRE(numLoadedCold >= expectedSuccessKeys.size());
+
+            auto preLoadPrefetchHitRate = root.getPrefetchHitRate();
+            REQUIRE(preLoadPrefetchHitRate == 0);
+
+            // Ensure that the correct entries were loaded.
+            for (auto const& k : expectedSuccessKeys)
+            {
+                auto txle = ltx2.load(k);
+                REQUIRE(txle);
+                REQUIRE(entrySet.find(txle.current()) != entrySet.end());
+            }
+            // Ensure that the meter is exhausted after loading.
```

Comment:
After the load loop, we need to check prefetch hit rate to make sure it's 100% to ensure that the keys we loaded in the loop were actually in the cache.